lib/puppet/transaction/report.rb

```diff
@@ -147,8 +147,10 @@ def as_logging_destination(&block)
   end
 
   # @api private
-  def <<(msg)
-    @logs << msg
+  def <<(log_entry)
+    # runtime errors after completion of report are added after report has finalized
+    @status = 'failed' if log_entry.level == :err
```

Comment:
Yeah, I would hope that an exception is raised in the transaction, and we could rescue it/set the report status explicitly.